cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -820,6 +998,58 @@ const FunctionDoc pow_checked_doc{
     ("An error is returned when integer to negative integer power is encountered,\n"
      "or integer overflow is encountered."),
     {"base", "exponent"}};
+
+const FunctionDoc bit_wise_not_doc{
+    "Bit-wise negate the arguments element-wise", ("This function never fails."), {"x"}};
```

Comment:
Is there any point in mentioning that the function never fails? This is true of several other functions as well.